### PR TITLE
Fix IPCHLE for gethostbyname.

### PR DESCRIPTION
**Pull request by magumagu**
It turns out that the offsets of the data in the return buffer matter: the
offset 0x340 is hardcoded into the code that uses the result.

With that fixed, get rid of the completely wrong 0x80000000 hack.

Thanks @tueidj for pointing out how this is supposed to work.

### CODE REVIEW

**Comment by tueidj** [comment]
Couple of other minor things:
- I'm not sure if they half-assed the implementation of aliases and no-one noticed because they're not used, or the address and alias pointers are meant to be one continuous NULL terminated list. The PPC code seems to think they're one list, which would mean detecting the number of addresses returned should require measuring the distance between h_addr_list and h_aliases instead of looking for the NULL.
- the maximum number of combined aliases and addresses returned is 71; (0x460 - 0x340) / sizeof(void*) = 72, minus one for the terminator entry. The output buffer could be overflowed if a host has too many addresses.

File: Source/Core/Core/IPC_HLE/WII_IPC_HLE_Device_net.cpp [hunk 1]
```diff
@@ -1057,50 +1057,47 @@ IPCCommandResult CWII_IPC_HLE_Device_net_ip_top::IOCtl(u32 _CommandAddress)
...
+					u32 addr = BufferOut + GETHOSTBYNAME_IP_PTR_LIST_OFFSET + i * 4;
+					Memory::Write_U32(BufferOut + GETHOSTBYNAME_IP_LIST_OFFSET + i * 4, addr);
 				}
+				Memory::Write_U32(0, BufferOut + GETHOSTBYNAME_IP_PTR_LIST_OFFSET + num_ip_addr * 4);
+
+				// Aliases - empty.
+				// TODO: Fill this in?  (Not that it's particularly useful...)
+				Memory::Write_U32(BufferOut + GETHOSTBYNAME_IP_PTR_LIST_OFFSET + num_ip_addr * 4, BufferOut + 4);
+				Memory::Write_U32(0, BufferOut + GETHOSTBYNAME_IP_PTR_LIST_OFFSET + num_ip_addr * 4 + 4);
+
```
Comment (tueidj) [inline Source/Core/Core/IPC_HLE/WII_IPC_HLE_Device_net.cpp]: This second NULL terminator isn't necessary, h_aliases already points to the first one.

**Comment by magumagu** [comment]
I'm guessing they half-assed the implementation; structure appears to be the Berkeley sockets hostent, which requires null-terminated lists.

**Comment by Sonicadvance1** [comment]
+1 I like how it cleans up that mess of code

**Comment by magumagu** [comment]
@dolphin-emu-bot rebuild

Updated to address review comments.

**Comment by magumagu** [comment]
@dolphin-emu-bot rebuild

**Pull request merged**